File: swift/3d_ex/main.swift
import Foundation

func err(_ exc: String) -> Never {
    fatalError(exc)
}

func initVal() -> Int {
    guard let line = readLine(), let val = Int(line) else {
        err("Invalid input")
    }
    return val
}

func countingDup(_ n: Int) -> Int {
    var a = initVal()
    var b = a
    var count = 0

    for _ in 1..<n {
        a = initVal()
        if a == b {
            count += 1
        }
        b = a
    }

    return count
}

let n = initVal()
if n <= 0 {
    err("Invalid input")
}

print(countingDup(n))

File: go/3d_ex/main.go
package main

import (
	"fmt"
)

func initVal() int {
	var val int
	_, err := fmt.Scanf("%d", &val)
	if err != nil {
		panic("Invalid input")
	}
	return val
}

func countingDup(n int) int {
	a := initVal()
	b := a
	count := 0

	for i := 1; i < n; i++ {
		a = initVal()
		if a == b {
			count++
		}
		b = a
	}

	return count
}

func main() {
	n := initVal()
	if n <= 0 {
		panic("Invalid input")
	}

	fmt.Println(countingDup(n))
}
